common/test/java/com/couchbase/lite/internal/core/C4DatabaseTest.kt

```diff
@@ -0,0 +1,728 @@
+            assertEquals(C4Constants.LiteCoreError.NOT_FOUND.toLong(), ex.code.toLong())
+        }
+    }
+
+    @Test
+    fun testDatabaseCopyFromNonexistentDbFails() {
+        try {
+            C4Database.copyDb(
+                File(c4Database.dbPath?.let { File(it).parentFile }, "x" + C4Database.DB_EXTENSION).path,
+                getScratchDirectoryPath(getUniqueName("c4_test_2")),
+                getUniqueName("c4_copy_test_db"),
+                flags,
+                C4Constants.EncryptionAlgorithm.NONE,
+                null)
+            fail("Copy from non-existent database should fail")
+        } catch (ex: LiteCoreException) {
+            assertEquals(C4Constants.ErrorDomain.LITE_CORE, ex.domain)
+            assertEquals(C4Constants.LiteCoreError.NOT_FOUND, ex.code)
+        }
+    }
+
+    @Test
+    @Throws(LiteCoreException::class)
+    fun testDatabaseCopyToExistingDBFails() {
+        createRev("doc001", REV_ID_1, fleeceBody)
+        createRev("doc002", REV_ID_1, fleeceBody)
+        val srcDbPath = c4Database.dbPath
+        val dstParentDirPath = getScratchDirectoryPath(getUniqueName("c4_test_2"))
+        var targetDb = C4Database.getDatabase(
+            dstParentDirPath,
+            dbName,
+            flags,
+            C4Constants.EncryptionAlgorithm.NONE,
+            null)
+        createRev(targetDb, "doc001", REV_ID_1, fleeceBody)
+        assertEquals(1, targetDb.documentCount)
+        targetDb.close()
+        try {
+            C4Database.copyDb(
+                srcDbPath!!,
+                dstParentDirPath,
+                dbName,
+                flags,
+                C4Constants.EncryptionAlgorithm.NONE,
+                null)
+        } catch (ex: LiteCoreException) {
+            assertEquals(C4Constants.ErrorDomain.POSIX, ex.domain)
+            assertEquals(POSIX_EEXIST, ex.code)
+        }
+        targetDb = C4Database.getDatabase(
+            dstParentDirPath,
+            dbName,
+            flags,
+            C4Constants.EncryptionAlgorithm.NONE,
+            null)
+        assertEquals(1, targetDb.documentCount)
+        targetDb.close()
+    }
+
+
+    // - Scopes and Collections
+
+    //  Test create 1 scope and 1 collection
+    @Test
+    fun testCreateScopeAndCollection() {
+        c4Database.addCollection("test_scope", "test_coll")
+        assertEquals(2, c4Database.scopeNames.size)
+        assertTrue(c4Database.hasScope("test_scope"))
+        assertEquals(setOf("_default", "test_scope"), c4Database.scopeNames)
+
+        assertEquals(1, c4Database.getCollectionNames("test_scope").size)
+        assertEquals(setOf("test_coll"), c4Database.getCollectionNames("test_scope"))
+    }
+
+    // Test create multiple collections in a scope
+    @Test
+    fun testCreateMultipleCollections() {
+        c4Database.addCollection("test_scope", "test_coll_1")
+        c4Database.addCollection("test_scope", "test_coll_2")
+
```

Comment:
Can you verify that the number of scopes in the database changes by the expected amount, after you add the collections?